sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/utils/HttpUtil.java

```diff
@@ -99,6 +99,28 @@ public static String post(String url, Map<String, String> headers, String body,
         return result;
     }
 
+    public static HttpResponse postWithResponse(String url, Map<String, String> headers, String body,
+        String contentType) {
+
+        HttpResponse result = null;
+
+        try (CloseableHttpClient client = buildClient()) {
+            HttpPost httpPost = new HttpPost(url);
+            httpPost.addHeader(USER_AGENT_KEY, USER_AGENT_VALUE);
+
+            if (headers != null) {
+                headers.forEach(httpPost::addHeader);
+                httpPost.addHeader("Content-Type", contentType);
+            }
+
+            httpPost.setEntity(new StringEntity(body, ContentType.create(contentType)));
+            result = client.execute(httpPost, createResponseHandlerForAuthChallenge());
+        } catch (IOException ioe) {
+            LOGGER.log(WARNING, "Unable to finish the http post request.", ioe);
```

Comment:
We should have retries for transient failures.

Reply:
We can address this on a future PR.